frontend/src/models/game.js

```diff
@@ -79,4 +81,29 @@ export class Game {
   resetPlayers() {
     this.players.forEach(player => player.reset());
   }
+
+  initializeAffinities() {
+    this.players.forEach(player => player.affinities.setPlayers(this.players));
+  }
+
+  affinityEvent(event, player) {
+    switch (event) {
+      case affinityEvents.announcement:
+        this.players.forEach(playerLoop => {
+          if (playerLoop.id !== player.id) {
+            playerLoop.affinities.declaresParty(player);
+          }
+        });
+        break;
+      case affinityEvents.queen_of_clubs:
+        this.players.forEach(playerLoop => {
+          if (playerLoop.id !== player.id) {
+            playerLoop.affinities.declaresParty(player);
+          }
+        });
+        break;
```

Comment:
> if `announcement` and `queen_of_clubs` are handled exactly the same, we could simplify here:

Your're right. This was because I was thinking further. But actually I don't think the behaviour here will change at all, let's take this :)